flox/core.py

```diff
@@ -1364,6 +1364,7 @@ def groupby_reduce(
     isbin: bool = False,
     axis=None,
     fill_value=None,
+    dtype=None,
```

Comment:
does this need a minimum numpy version?

Reply:
https://numpy.org/devdocs/reference/typing.html#numpy.typing.DTypeLike Says it was introduced in 1.20. 
Minimum here is 1.19 right? Then it might not be worth it. :/

Reply:
OK maybe we should bump it given we would also get the `like` kwarg (see #132)